flink-ai-flow/ai_flow_plugins/job_plugins/flink/flink_job_plugin.py

```diff
@@ -0,0 +1,328 @@
+    contexts: List[ExecutionContext] = []
+    exec_env, table_env, statement_set = flink_env.create_env()
+    for index in range(len(run_graph.nodes)):
+        caller: FlinkPythonExecutor = serialization_utils.deserialize(run_graph.executor_bytes[index])
+        executors.append(caller)
+        node: AINode = run_graph.nodes[index]
+        execution_context = ExecutionContext(config=node.node_config,
+                                             job_execution_info=job_execution_info,
+                                             execution_env=exec_env,
+                                             table_env=table_env,
+                                             statement_set=statement_set)
+        contexts.append(execution_context)
+
+    def setup():
+        for ii in range(len(executors)):
+            cc = executors[ii]
+            cc.setup(contexts[ii])
+
+    def close():
+        for ii in range(len(executors)):
+            cc = executors[ii]
+            cc.close(contexts[ii])
+
+    setup()
+    value_map = {}
+    for i in range(len(run_graph.nodes)):
+        node = run_graph.nodes[i]
+        c = executors[i]
+        if node.instance_id in run_graph.dependencies:
+            ds = run_graph.dependencies[node.instance_id]
+            params = []
+            for d in ds:
+                params.append(value_map[d.tail][d.port])
+            value_map[node.instance_id] = c.execute(contexts[i], params)
+        else:
+            value_map[node.instance_id] = c.execute(contexts[i], [])
+    close()
+    job_client = statement_set.execute().get_job_client()
+    if job_client is not None:
+        with open('./job_id', 'w') as fp:
+            print('job_id {}'.format(job_client.get_job_id()))
+            fp.write(str(job_client.get_job_id()))
+        job_client.get_job_execution_result(user_class_loader=None).result()
+
+
+class FlinkJob(Job):
+    def __init__(self, job_config: JobConfig):
+        super().__init__(job_config)
+        self.run_graph_file: Text = None
+        self.flink_env_file: Text = None
+        self.stdout_log: Text = None
+        self.stderr_log: Text = None
+
+
+class FlinkJobHandler(JobHandler):
+
+    def __init__(self, job: Job,
+                 job_execution: JobExecutionInfo):
+        super().__init__(job=job, job_execution=job_execution)
+        self.sub_process = None
+        self.run_args_file = None
+
+    def wait_finished(self):
+        self.log.info('Output:')
+
+        self.sub_process.wait()
+
+        self.log.info('Command exited with return code %s', self.sub_process.returncode)
+
+        if self.sub_process.returncode != 0:
+            raise Exception('Flink run failed. The command returned a non-zero exit code.')
+        return None
+
+
+class FlinkJobPlugin(AbstractJobPlugin, ABC):
+
+    def __init__(self) -> None:
+        super().__init__()
+
+    @classmethod
+    def build_run_graph(cls, sub_graph: AISubGraph) -> RunGraph:
```

Comment:
This method is same as `build_run_graph` in `PythonJobPlugin`, why not make it a common one